### PR TITLE
feat(testing): add a slider element

### PR DESCRIPTION
**Pull request by matthieu-crouzet**
## Proposed change

Add a slider element

## Related issues

- :rocket: Feature #2045

### CODE REVIEW

**Review by cpaulve-1A — DISMISSED** [review]

File: packages/@o3r/testing/src/core/angular-materials/slider-material.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { O3rElement } from '../element';
+import { SliderElementProfile } from '../elements';
```
Comment (kpanot) [inline packages/@o3r/testing/src/core/angular-materials/slider-material.ts:2]:
```suggestion
import type { SliderElementProfile } from '../elements';
```

File: packages/@o3r/testing/src/core/angular-materials/slider-material.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ */
+export interface MatSliderProfile extends SliderElementProfile {}
+
+/**
+ * Mock for ElementProfile class.
+ * This class is used for fixture compilation purpose.
+ */
+export class MatSlider extends O3rElement implements MatSliderProfile {
+  constructor(_sourceElement: any) {
+    super(_sourceElement);
```
Comment (kpanot) [inline packages/@o3r/testing/src/core/angular-materials/slider-material.ts:17]:
```suggestion
  constructor(sourceElement: any) {
    super(sourceElement);
```

File: packages/@o3r/testing/src/core/angular/angular-materials/slider-material.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { DebugElement } from '@angular/core';
+import { MatSliderProfile } from '../../angular-materials';
```
Comment (kpanot) [inline packages/@o3r/testing/src/core/angular/angular-materials/slider-material.ts:2]:
```suggestion
import type { MatSliderProfile } from '../../angular-materials';
```

File: packages/@o3r/testing/src/core/angular/angular-materials/slider-material.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+export { SelectElementProfile } from '../../elements';
+
+/**
+ * Implementation dedicated to angular / TestBed.
+ */
+export class MatSlider extends O3rSliderElement implements MatSliderProfile {
+  constructor(sourceElement: DebugElement | O3rElement) {
+    super(
+      sourceElement instanceof O3rElement ? sourceElement.sourceElement : sourceElement,
+      '.mdc-slider__track',
```
Comment (kpanot) [inline packages/@o3r/testing/src/core/angular/angular-materials/slider-material.ts:15]: Could be use from `cssClasses.TRACK` (from `import { cssClasses } from '@material/slider';`)

File: packages/@o3r/testing/src/core/angular/elements/slider-element.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
+import { DebugElement } from '@angular/core';
+import { By } from '@angular/platform-browser';
+import { SliderElementProfile } from '../../elements';
```
Comment (kpanot) [inline packages/@o3r/testing/src/core/angular/elements/slider-element.ts:3]:
```suggestion
import type { SliderElementProfile } from '../../elements';
```

File: packages/@o3r/testing/src/core/playwright/angular-materials/slider-material.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import {MatSliderProfile} from '../../angular-materials';
```
Comment (kpanot) [inline packages/@o3r/testing/src/core/playwright/angular-materials/slider-material.ts:1]:
```suggestion
import type { MatSliderProfile } from '../../angular-materials';
```

File: packages/@o3r/testing/src/core/playwright/elements/slider-element.ts [hunk 7]
```diff
@@ -0,0 +1,94 @@
+import {SliderElementProfile} from '../../elements';
```
Comment (kpanot) [inline packages/@o3r/testing/src/core/playwright/elements/slider-element.ts:1]:
```suggestion
import type { SliderElementProfile } from '../../elements';
```

File: packages/@o3r/testing/src/core/playwright/angular-materials/slider-material.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import type { cssClasses } from '@material/slider';
+import type { MatSliderProfile } from '../../angular-materials';
+export type { SliderElementProfile } from '../../elements';
```
Comment (kpanot) [inline packages/@o3r/testing/src/core/playwright/angular-materials/slider-material.ts:3]: should be removed

**Review by kpanot — APPROVED** [review]

**Pull request merged**